jax/_src/interpreters/mlir.py

```diff
@@ -1399,6 +1453,113 @@ def get_lowering(primitive: core.Primitive) -> LoweringRule | None:
     core.clean_up_dead_vars(eqn, env, last_used)
   return map(read, jaxpr.outvars), tokens
 
+def lower_multi_platform(ctx: LoweringRuleContext,
```

Comment:
I don't have any concrete suggestions, but I found it a bit confusing that a function called `lower_multi_platform` handles both the single- and multi- platform cases.

Reply:
I have kept for now the old lowering rule selection code just to make it clearer that the scope of this change is limited. But eventually lower_multi_platform is going to be the only selection code. I think it make sense for now to call it like that. It does handle single platform selection code as a special case.